tensorflow_datasets/structured/heart_disease.py

```diff
@@ -0,0 +1,176 @@
+year = "1988",
+title = "{UCI} Machine Learning Repository",
+url = "http://archive.ics.uci.edu/ml/datasets/Heart+Disease",
+institution = "University of California, Irvine, School of Information and Computer Sciences"
+}
+"""
+
+_DESCRIPTION = """\
+This data set contain 13 attributes and labels of heart disease from \
+303 participants from Cleveland since Cleveland data was most commonly\
+used in modern research.
+
+Attribute by column index
+1. age      : age in years
+2. sex      : sex (1 = male; 0 = female)
+3. cp       : chest pain type
+    (1 = typical angina; 2 = atypical angina; 3 = non-anginal pain; 4 = asymptomatic)
+4. trestbps : resting blood pressure (in mm Hg on admission to the hospital)
+5. chol     : serum cholestoral in mg/dl
+6. fbs      : (fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)
+7. restecg  : resting electrocardiographic results
+8. thalach  : maximum heart rate achieved
+9. exang    : exercise induced angina (1 = yes; 0 = no)
+10. oldpeak : ST depression induced by exercise relative to rest
+11. slope   : the slope of the peak exercise ST segment (1 = upsloping; 2 = flat; 3 = downsloping)
+12. ca      : number of major vessels (0-3) colored by flourosopy
+13. thal    : 3 = normal; 6 = fixed defect; 7 = reversable defect
+14. num (the predicted attribute): diagnosis of heart disease (angiographic disease status)
+    (0 = < 50% diameter narrowing, no presence of heart disease;
+     1 = > 50% diameter narrowing, with increasing severity)
+Dataset Homepage: http://archive.ics.uci.edu/ml/datasets/Heart+Disease
+"""
+
+_SEX_DICT = collections.OrderedDict([
+    ("1.0", "male"), ("0.0", "female")
+])
+_CP_DICT = collections.OrderedDict([
+    ("1.0", "typical angina"), ("2.0", "atypical angina"),
+    ("3.0", "non-anginal pain"), ("4.0", "asymptomatic")
+])
+_SLOPE_DICT = collections.OrderedDict([
+    ("1.0", "upsloping"), ("2.0", "flat"), ("3.0", "downsloping")
+])
+_EXE_DICT = collections.OrderedDict([
+    ("0.0", "no"), ("1.0", "yes")
+])
+_FBS_DICT = collections.OrderedDict([
+    ("0.0", "false"), ("1.0", "true")
+])
+_THAL_DICT = collections.OrderedDict([
+    ("3.0", "normal"), ("6.0", "fixed defect"), ("7.0", "reversable defect")
+])
+
+_CP_NAMES = ['typical angina', 'atypical angina',
+             'non-anginal pain', 'asymptomatic']
+_SLOPE_NAMES = ['upsloping', 'flat', 'downsloping']
+_THAL_NAMES = ['normal', 'fixed defect', 'reversable defect']
+
+_DOWNLOAD_URL = 'http://archive.ics.uci.edu/ml/machine-learning-databases/heart-disease/processed.cleveland.data'
+
+def convert_to_label(d, dictionary):
+  return dictionary[d]
+
+def convert_to_int(d):
+  return int(float(d))
+
+def convert_to_float(d):
+  return float(d)
+
+def return_same(d):
+  return d
+
+FEATURE_DICT = collections.OrderedDict([
+    ("age", (tf.int32, convert_to_int)),
+    ("sex", (tfds.features.ClassLabel(names=['female', 'male']),
+             lambda d: convert_to_label(d, _SEX_DICT))),
```

Comment:
This seems to complexify the implementation. Why not:
```
features = FeatureDict({
    'features': {
        'age': tf.int32,
        'sex': tfds.features.ClassLabel(names=['female', 'male']),
        ...,
    }
})
```

Then:
```
yield key, {
    'features': {
        'age': int(values['ages']),
        'sex': int(values['sex']),
        ...
        'slope': int(values['slope']) - 1
    }
}
```

Reply:
Done.